stdlib/csv/from.go

```diff
@@ -138,21 +139,29 @@ func (c *CSVSource) Run(ctx context.Context) {
 	var err error
 	var max execute.Time
 	maxSet := false
+
 	for _, t := range c.ts {
 		// For each downstream transformation, instantiate a new result
 		// decoder. This way a table instance goes to one and only one
 		// transformation. Unlike other sources, tables from csv sources
 		// are not read-only. They contain mutable state and therefore
 		// cannot be shared among goroutines.
-		decoder := csv.NewResultDecoder(csv.ResultDecoderConfig{
+		decoder := csv.NewMultiResultDecoder(csv.ResultDecoderConfig{
 			Allocator: c.alloc,
 			Context:   ctx,
 		})
-		result, decodeErr := decoder.Decode(strings.NewReader(c.tx))
+		results, decodeErr := decoder.Decode(ioutil.NopCloser(strings.NewReader(c.tx)))
```

Comment:
You need to remember to defer a release to ensure it closes. I don't think it matters for this usage though so just good practice.